niworkflows/viz/plots.py

```diff
@@ -239,76 +241,75 @@ def plot_carpet(
-        interpolation="none",
-        aspect="auto",
-        cmap=cmap
-    )
-
-    ax0.grid(False)
-    ax0.spines["left"].set_visible(False)
-    ax0.spines["bottom"].set_color("none")
-    ax0.spines["bottom"].set_visible(False)
-
     # Calculate time decimation factor
     t_dec = max(int((1.8 * n_trs) // size[1]), 1)
-    data = data[np.hstack(list(segments.values())), drop_trs::t_dec]
-
-    # Carpet plot
-    ax1 = plt.subplot(gs[1])
-    ax1.imshow(
-        data,
-        interpolation="nearest",
-        aspect="auto",
-        cmap="gray",
-        vmin=vminmax[0],
-        vmax=vminmax[1],
-    )
+    data = data[:, drop_trs::t_dec]
 
-    ax1.grid(False)
-    ax1.set_yticks([])
-    ax1.set_yticklabels([])
-
-    xticks = np.linspace(0, data.shape[-1], endpoint=True, num=7)
-    xlabel = "time-points (index)"
-    xticklabels = (xticks * n_trs / data.shape[-1]).astype("uint32") + drop_trs
-    if tr is not None:
-        xlabel = "time (mm:ss)"
-        xticklabels = [
-            f"{int(t // 60):02d}:{(t % 60).round(0).astype(int):02d}"
-            for t in (tr * xticklabels)
-        ]
+    nsegments = len(segments)
+
+    # Define nested GridSpec
+    gs = mgs.GridSpecFromSubplotSpec(
+        nsegments,
+        1,
+        subplot_spec=subplot,
+        hspace=0.05,
+        height_ratios=[len(v) for v in segments.values()]
+    )
 
-    ax1.set_xticks(xticks)
-    ax1.set_xlabel(xlabel)
-    ax1.set_xticklabels(xticklabels)
+    for i, (label, indices) in enumerate(segments.items()):
+        # Carpet plot
+        ax = plt.subplot(gs[i])
+
+        ax.imshow(
+            data[indices, :],
+            interpolation="nearest",
+            aspect="auto",
+            cmap="gray",
+            vmin=vminmax[0],
+            vmax=vminmax[1],
+        )
 
-    # Remove and redefine spines
-    for side in ["top", "right"]:
         # Toggle the spine objects
-        ax0.spines[side].set_color("none")
-        ax0.spines[side].set_visible(False)
-        ax1.spines[side].set_color("none")
-        ax1.spines[side].set_visible(False)
-
-    ax1.yaxis.set_ticks_position("left")
-    ax1.xaxis.set_ticks_position("bottom")
-    ax1.spines["bottom"].set_visible(False)
-    ax1.spines["left"].set_color("none")
-    ax1.spines["left"].set_visible(False)
-    if title:
-        ax1.set_title(title)
+        ax.spines["top"].set_color("none")
+        ax.spines["top"].set_visible(False)
+        ax.spines["right"].set_color("none")
+        ax.spines["right"].set_visible(False)
+
+        # Make colored left axis
+        ax.spines["left"].set_linewidth(3)
+        ax.spines["left"].set_color(colors[i])
+        ax.spines["left"].set_capstyle("butt")
+        ax.spines["left"].set_position(("outward", 2))
+
+        # Make all subplots have same xticks
+        xticks = np.linspace(0, data.shape[-1], endpoint=True, num=7)
+        ax.set_xticks(xticks)
+        ax.set_yticks([])
+        ax.set_ylabel(label)
```

Comment:
Can you color code the labels with the same color as the colorbar ? It will be more easy to read.
```suggestion
        ax.set_ylabel(label, color = colors[i])
```

Reply:
Are you sure?? I think that will actually make them harder to read.

Reply:
Let's go with them in black and you can try to convince me whenever you have time to generate some colored examples :)